### ISSUE TITLE
ISchedulerListener is firing twice for events.

### ISSUE THREAD

**Issue by DRAirey1** [quartznet/quartznet#2635]
I have a simple scheduler that I'm loading from an SQL table (that resembles the Quartz records):

```
            // Create, configure and start the scheduler.
            this.scheduler = await this.schedulerFactory.GetScheduler(cancellationToken);
            this.scheduler.JobFactory = this.jobFactory;
            this.scheduler.ListenerManager.AddSchedulerListener(this.schedulerListener);
            await this.scheduler.Start(cancellationToken).ConfigureAwait(false);

```

I've verified with logging that this code is only run once. I've verified with logging that the job is only running once.
The listener is a trivial implementation of an IScheduleListener:

```
        /// <inheritdoc/>
        public async Task JobAdded(IJobDetail jobDetail, CancellationToken cancellationToken = default)
        {
            this.logger.LogInformation($"Starting ({jobDetail.Key.Group}, {jobDetail.Key.Name}).");
        }
        

```
When I run this program, I get:

```
info: ThetaRex.Market.Jobs.JobService[0]
      Starting (ETrade, Start Quote Service Job).
info: ThetaRex.Market.Jobs.JobService[0]
      Starting (ETrade, Start Quote Service Job).

```

The listener task is getting called twice for every job I add and twice for every job that is deleted.

**Comment by DRAirey1** [comment]
Figured it out. I wanted to add a SchedulerListener, so I followed the instructions I found in a tutorial. However, what the tutorial failed to mention is that if you already provide a transient for an ISchedulerListener, it's automatically added to the JobScheduler, so when I added this explicit instruction:

`            this.scheduler.ListenerManager.AddSchedulerListener(schedulerListener);
`
I was duplicating the number of listeners, hence the listener was firing twice.

**Comment by DRAirey1** [comment]
Figured it out. I wanted to add a SchedulerListener, so I followed the instructions I found in a tutorial. However, what the tutorial failed to mention is that if you already provide a transient for an ISchedulerListener, it's automatically added to the JobScheduler, so when I added this explicit instruction:

`            this.scheduler.ListenerManager.AddSchedulerListener(schedulerListener);
`
I was duplicating the number of listeners, hence the listener was firing twice.
